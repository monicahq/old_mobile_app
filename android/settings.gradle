rootProject.name = 'Monica'
include ':react-native-vector-icons'
project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
include ':react-native-splash-screen'
project(':react-native-splash-screen').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-splash-screen/android')
include ':react-native-device-info'
project(':react-native-device-info').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-device-info/android')
include ':react-native-code-push'
project(':react-native-code-push').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-code-push/android/app')
include ':appcenter-crashes'
project(':appcenter-crashes').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter-crashes/android')
include ':appcenter-analytics'
project(':appcenter-analytics').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter-analytics/android')
include ':appcenter'
project(':appcenter').projectDir = new File(rootProject.projectDir, '../node_modules/appcenter/android')

include ':app'
