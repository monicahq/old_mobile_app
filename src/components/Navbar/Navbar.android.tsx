import React, {PureComponent} from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {commonStyles} from '@theme';
import {INavbarProps} from './Navbar.props';

export class Navbar extends PureComponent<INavbarProps> {
  public render() {
    const {onBack, title} = this.props;

    return (
      <Icon.ToolbarAndroid
        logo={!onBack ? require('@assets/logo.png') : undefined}
        navIconName={onBack && 'arrow-back'}
        onIconClicked={onBack}
        title={onBack ? title : `  ${title}`}
        titleColor="white"
        onActionSelected={this.onActionSelected}
        style={commonStyles.toolbarAndroid}
        actions={this.getActions()}
      />
    );
  }

  private onActionSelected = () => {
    const {rightAction} = this.props;
    if (typeof rightAction === 'function') {
      rightAction();
    }
  };

  private getActions = () => {
    const {rightTitle, rightIcon} = this.props;

    return rightTitle
      ? [
          {
            title: rightTitle,
            iconName: rightIcon,
            iconSize: 30,
            show: 'always',
          },
        ]
      : [];
  };
}
