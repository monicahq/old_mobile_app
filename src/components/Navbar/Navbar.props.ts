import {ReactNode} from 'react';

export interface INavbarProps {
  title: string | ReactNode;
  onBack?: any;
  rightTitle?: string;
  rightAction?: (...args) => void;
  rightIcon?: string;
}
