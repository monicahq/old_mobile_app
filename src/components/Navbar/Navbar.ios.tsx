import React, {PureComponent} from 'react';
import {TouchableOpacity} from 'react-native';
import NavigationBar from 'react-native-navbar';
import Icon from 'react-native-vector-icons/Ionicons';

import {hitSlop, navbarColor, statusBarLightContent} from '@theme';
import {INavbarProps} from './Navbar.props';
import {styles} from './Navbar.styles';

export class Navbar extends PureComponent<INavbarProps> {
  public render() {
    const {onBack, title} = this.props;

    return (
      <NavigationBar
        style={styles.navbar}
        rightButton={this.getRightButton()}
        leftButton={
          onBack && (
            <TouchableOpacity
              onPress={onBack}
              style={styles.leftButton}
              hitSlop={hitSlop}
            >
              <Icon name="ios-arrow-back" size={30} color="white" />
            </TouchableOpacity>
          )
        }
        title={typeof title === 'string' ? {title, tintColor: 'white'} : title}
        tintColor={navbarColor}
        statusBar={statusBarLightContent}
      />
    );
  }
  private getRightButton() {
    const {rightAction, rightTitle} = this.props;
    if (!rightTitle) {
      return null;
    }
    return {
      title: rightTitle,
      handler: rightAction,
      tintColor: 'white',
    };
  }
}
