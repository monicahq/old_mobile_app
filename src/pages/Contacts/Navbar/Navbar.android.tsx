import React, {PureComponent} from 'react';
import SearchBar from 'react-native-searchbar';
import Icon from 'react-native-vector-icons/MaterialIcons';

import {NetworkStatusContainer} from '@containers/NetworkStatus';
import {I18n} from '@i18n';
import {commonStyles} from '@theme';
import {INavbarProps} from './Navbar.props';

const actions = [
  {
    title: I18n.t('common:search'),
    iconName: 'search',
    iconSize: 30,
    show: 'always',
  },
];

export class Navbar extends PureComponent<INavbarProps, {}> {
  private searchBar;

  public onActionSelected = () => {
    this.searchBar.show();
  };
  public onBack = () => {
    this.searchBar.hide();
    this.props.onSearchTextChanged('');
  };
  public hide = () => {
    this.searchBar.hide();
  };
  public searchBarRef = ref => {
    this.searchBar = ref;
  };
  public render() {
    return [
      <Icon.ToolbarAndroid
        logo={require('@assets/logo.png')}
        key={0}
        title={'  ' + I18n.t('contacts:contacts')}
        titleColor="white"
        onActionSelected={this.onActionSelected}
        style={commonStyles.toolbarAndroid}
        actions={actions}
      />,
      <SearchBar
        key={1}
        ref={this.searchBarRef}
        placeholder={I18n.t('contacts:search')}
        onBlur={this.hide}
        handleSearch={this.props.onSearchTextChanged}
        onBack={this.onBack}
      />,
      <NetworkStatusContainer key={2} />,
    ];
  }
}
