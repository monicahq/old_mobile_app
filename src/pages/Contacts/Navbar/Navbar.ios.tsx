import React, {PureComponent} from 'react';
import {Image} from 'react-native';
import NavigationBar from 'react-native-navbar';
import SearchBar from 'react-native-search-bar';

import {I18n} from '@i18n';

import {NetworkStatusContainer} from '@containers/NetworkStatus';
import {navbarColor, statusBarLightContent} from '@theme';
import {INavbarProps} from './Navbar.props';
import {styles} from './Navbar.styles';

export class Navbar extends PureComponent<INavbarProps, {}> {
  public render() {
    return [
      <NavigationBar
        key={0}
        leftButton={
          <Image
            source={require('@assets/logo.png')}
            style={styles.navbarLogo}
          />
        }
        title={
          <SearchBar
            placeholder={I18n.t('contacts:search')}
            hideBackground={true}
            showsCancelButtonWhileEditing={false}
            autoCapitalize="words"
            searchBarStyle="minimal"
            textColor="#ACACAC"
            onChangeText={this.props.onSearchTextChanged}
            style={styles.searchBar}
          />
        }
        rightButton={<NetworkStatusContainer />}
        tintColor={navbarColor}
        statusBar={statusBarLightContent}
      />,
    ];
  }
}
