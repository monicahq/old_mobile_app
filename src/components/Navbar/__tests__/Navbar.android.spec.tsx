import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Navbar} from '../Navbar.android';

describe('Components', () => {
  describe('Navbar Android', () => {
    let onBack;
    let onRight;

    beforeEach(() => {
      onBack = jest.fn();
      onRight = jest.fn();
    });

    it('should renders correctly', () => {
      expect(
        toJson(shallow(<Navbar onBack={onBack} title="My title" />))
      ).toMatchSnapshot();

      expect(toJson(shallow(<Navbar title="My title" />))).toMatchSnapshot();
    });

    it('onActionSelected should trigger the right action', () => {
      const tree = shallow(
        <Navbar onBack={onBack} rightAction={onRight} title="My title" />
      );
      tree.instance().onActionSelected();
      expect(onRight.mock.calls.length).toBe(1);
      expect(onRight.mock.calls[0]).toEqual([]);
    });
  });
});
