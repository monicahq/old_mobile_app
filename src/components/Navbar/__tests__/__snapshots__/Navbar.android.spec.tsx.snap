// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Navbar Android should renders correctly 1`] = `
<IconToolbarAndroid
  actions={Array []}
  iconSize={24}
  navIconName="arrow-back"
  onActionSelected={[Function]}
  onIconClicked={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#355974",
      "height": 56,
    }
  }
  title="My title"
  titleColor="white"
/>
`;

exports[`Components Navbar Android should renders correctly 2`] = `
<IconToolbarAndroid
  actions={Array []}
  iconSize={24}
  logo={1}
  onActionSelected={[Function]}
  style={
    Object {
      "backgroundColor": "#355974",
      "height": 56,
    }
  }
  title="  My title"
  titleColor="white"
/>
`;
